jep/0000/README.adoc

```diff
@@ -0,0 +1,187 @@
+//| Requires
+//| :bulb: JEP-NUMBER, JEP-NUMBER... :bulb:
+//
+//
+// Uncomment and fill if this JEP is rendered obsolete by a later JEP
+//| Superseded-By
+//| :bulb: JEP-NUMBER :bulb:
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+== Abstract
+
+Running Jenkins on Kubernetes cluster is not trivial work. Kubernetes platform was released ten years after first
+version of Hudson project. It means Jenkins couldn't be designed to run on top of it.
+
+Jenkins Operator helps to deploy, configure and manage Jenkins on Kubernetes clusters in an immutable, scalable, secure way.
+
+== Specification
+
+This specification describes the problem, and the overall proposed solution,
+as it relates to the Jenkins project and existing structures.
+
+The details of the exact behavior and implementation of most areas will be addressed in one or more separate JEPs.
+
+=== Target Audience
+
+Jenkins Operator is targeted at existing and new Jenkins users (system administrators, DevOps engineers or regular programmers)
+who are not necessarily interested in becoming Jenkins experts, but rather wish to use Jenkins as a means to an end.
+That end being: *building, testing, and delivering their projects*, with as *little overhead* as possible.
+Potential users wish to use a stable and reliable CI/CD tool.
+
+Jenkins Operator is going to be the best solution for everybody who is interested in running Jenkins
+on Kubernetes cluster.
+
+=== Goals
+
+==== Run Jenkins on top of Kubernetes
+
+Use link:https://coreos.com/blog/introducing-operator-framework[operator pattern] to manage complex Jenkins stateful
+application on top of Kubernetes.
+
+==== Jenkins configuration in one place
+
+Put all information needed to configure Jenkins in a single
+link:https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/[CRD] Kubernetes resource.
+
+==== Keep Jenkins instance configured
+
+Watch all changes made to Jenkins configuration
+link:https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/[CR] and apply it to the live
+instance.
+
+==== Make Jenkins immutable
+
+Jenkins master instance should be restorable to the desired (configured) state at any point in time.
+Any changes made to the Jenkins configuration that are *not* in
+link:https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/[CR]
+should be removed after Jenkins master restart to *avoid snowflake* problem.
+
+==== Scale Jenkins agents
+
+Spawn agents on demand to scale Jenkins workloads.
+
+==== Make default configuration secure
+
+Provide secure configuration as the default option.
+
+==== Make backup of Jenkins jobs history
+
+Backup Jenkins jobs history using tools provided by Kubernetes itself and optionally a cloud provider.
```

Comment:
I think you are talking about _build_ history here: `$JENKINS_HOME/jobs/**/builds/`.

_Job_ history would track changes to `$JENKINS_HOME/jobs/**/config.xml`, something done in traditional Jenkins installations using the `jobConfigHistory` or `scm-sync-configuration` plugins, which ought to be superseded by a proper operator.

Reply:
@jglick please look at new commit